src/plugins/hetero/src/compiled_model.cpp

```diff
@@ -0,0 +1,762 @@
+                            cyclicInputsDependencies.emplace(input);
+                        }
+                    }
+                    for (auto&& input : node->inputs()) {
+                        auto& inputNodeSubgraphCyclicInputDependency =
+                            nodeSubgraphCyclicInputDependencies[InputNode(input)];
+                        auto& inputNodeSubgraphInputDependency = nodeSubgraphInputDependencies[InputNode(input)];
+                        if (!InferenceEngine::details::Intersects(nodeSubgraphCyclicInputDependency,
+                                                                  inputNodeSubgraphCyclicInputDependency) &&
+                            InferenceEngine::details::Intersects(cyclicInputsDependencies,
+                                                                 inputNodeSubgraphInputDependency)) {
+                            subgraphInputs.insert(input);
+                        }
+                    }
+                }
+            }
+        }
+
+        auto subgraphIds = CollectSubgraphs();
+
+        if (dumpDotFile) {
+            std::map<std::string, int> map_id;
+            for (auto&& v : subgraphIds) {
+                map_id.emplace(v.first->get_friendly_name(), v.second);
+            }
+            ov::hetero::debug::dump_subgraphs(model, queryNetworkResult, map_id);
+        }
+
+        // Break graph using insertion of result parameter split
+        NodeMap<ov::Node*> subgraphParameterToPrevResult;
+        std::vector<std::shared_ptr<ov::op::v0::Result>> results;
+        {
+            std::set<ov::Output<ov::Node>> subgraphOutputs;
+            for (auto&& input : subgraphInputs) {
+                if (!ov::op::util::is_parameter(input.get_node()) && !ov::op::util::is_constant(input.get_node())) {
+                    subgraphOutputs.insert(input.get_source_output());
+                }
+            }
+            for (auto&& output : subgraphOutputs) {
+                auto output_subgraph_id = subgraphIds.at(output.get_node());
+                auto inputs = output.get_target_inputs();
+                // Collect input subsets from other subgraphs. Each subset of inputs belongs to the same subgraph
+                std::map<int, std::set<ov::Input<ov::Node>>> input_subsets;
+                for (auto&& input : inputs) {
+                    auto input_subgraph_id = subgraphIds.at(input.get_node());
+                    if (output_subgraph_id != input_subgraph_id) {
+                        input_subsets[input_subgraph_id].emplace(input);
+                    }
+                }
+                // for each subset of inputs create separate Result operation if subset belongs to other
+                for (auto&& input_subset : input_subsets) {
+                    auto result = std::make_shared<ov::op::v0::Result>(output);
+                    result->set_friendly_name(output.get_node()->get_friendly_name() + "_" +
```

Comment:
Why do you create create new params/results with custom rules?

Reply:
https://github.com/openvinotoolkit/openvino/pull/18429/files#r1259448540